packages/design-system/src/components/web-components/ds-text-field/ds-text-field.test.tsx

```diff
@@ -0,0 +1,134 @@
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import './ds-text-field';
+
+/* eslint-disable @typescript-eslint/no-namespace */
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'ds-text-field': any;
+    }
+  }
+}
+/* eslint-enable */
+
+function renderTextField(props = {}) {
+  return render(<ds-text-field {...props} />);
+}
+
+describe('ds-text-field', function () {
+  it('renders text-field', () => {
+    const { asFragment } = renderTextField({ id: 'static-id' });
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('renders disabled input', () => {
+    renderTextField({ disabled: 'true' });
+    const input = screen.getByRole('textbox');
+    expect(input).toHaveAttribute('disabled');
+  });
+
+  it('does not render disabled when passed false', () => {
+    renderTextField({ disabled: 'false' });
+    const input = screen.getByRole('textbox');
+    expect(input).not.toHaveAttribute('disabled');
+  });
+
+  it('applies additional classes to the wrapper', () => {
```

Comment:
The following is probably more philosophical than the scope of this ticket warrants. Feel free to disregard my stream-of-conscious take here:

This test initially confused me because I associate "wrapper" with the host for 🚾, and having a special prop to apply a class to a host is unnecessary because it's HTML and `class` would work fine. 

Anyways... I had to do a double-take and read what the test was actually doing and realized "wrapper" means something different because it's querying the clearfix style in the test, which is the internal `div` within the host wrapper. Is this `div` even needed? Who clearfixes in the era of flexbox and grid?

Do we need better language when talking about these sorts of tags?

Reply:
We might very well be able to remove that "wrapper" div now. I agree that it's beyond the scope here, but it's worth considering for the future (@kim-cmsds).

That's a great point about the wording here being confusing. I think I'll change it to "root element" or something like that.